test/integration/ads_integration_test.cc

```diff
@@ -675,6 +675,85 @@ TEST_P(AdsIntegrationTest, RdsAfterLdsWithRdsChange) {
   makeSingleRequest();
 }
 
+// Regression test for the use-after-free crash when a listener awaiting an RDS update is destroyed
```

Comment:
Can you move this to another PR? I think it doesn't logically belong with this. Ideally, the changes in this PR are all mechanical.

Reply:
I could, and you're right, these are mostly mechanical changes ... but I want to try to push back a little. The fix for #6116 is in this PR, and I feel like it's a little weird for the regression test to land separately.